Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm

```diff
@@ -128,57 +107,63 @@ String convertEnumerationToString(MediaPlayerPrivateMediaSourceAVFObjC::SeekStat
     m_spatialTrackingLabel = player->spatialTrackingLabel();
 #endif
 
-    // addPeriodicTimeObserverForInterval: throws an exception if you pass a non-numeric CMTime, so just use
-    // an arbitrarily large time value of once an hour:
-    __block WeakPtr weakThis { *this };
-    m_timeJumpedObserver = [m_synchronizer addPeriodicTimeObserverForInterval:PAL::toCMTime(MediaTime::createWithDouble(3600)) queue:mainDispatchQueueSingleton() usingBlock:^(CMTime time) {
-#if LOG_DISABLED
-        UNUSED_PARAM(time);
+    m_renderer->notifyWhenErrorOccurs([weakThis = WeakPtr { *this }](PlatformMediaError) {
+        if (RefPtr protectedThis = weakThis.get()) {
+            protectedThis->setNetworkState(MediaPlayer::NetworkState::DecodeError);
+            protectedThis->setReadyState(MediaPlayer::ReadyState::HaveNothing);
+        }
+    });
+
+    if (RefPtr protectedPlayer = player) {
```

Comment:
Why is it possible for `player` to be null? This is a bigger change, but `MediaPlayerFactory` should take a `MediaPlayer&` to guarantee to the caller that `player` is non-null.

Reply:
this is mostly to satisfy the safety static analyser. I could assert here too.